File: Gixor-Backend/Gixor-Api/Gixor-Api-GitHub/src/main/java/com/cpy3f2/Gixor/Domain/Issue.java
package com.cpy3f2.Gixor.Domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * @author : simon
 * @description :
 * @last : 2024-11-04 00:35
 * Copyright (c) 2024. 保留所有权利。
 * @since : 2024-11-04 00:35
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class Issue {
    private Long id;
    private Integer number;
    private String state;
    private String title;
    private String body;
    private GitHubUser user;
    private List<Label> labels;
    private GitHubUser assignee;
    private List<GitHubUser> assignees;
    private Milestone milestone;
    private Boolean locked;
    @JsonProperty("active_lock_reason")
    private String activeLockReason;
    private Integer comments;
    @JsonProperty("closed_at")
    private LocalDateTime closedAt;
    @JsonProperty("created_at")
    private LocalDateTime createdAt;
    @JsonProperty("updated_at")
    private LocalDateTime updatedAt;
    private GitHubRepository repository;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Label {
        private Long id;
        private String name;
        private String description;
        private String color;
        private Boolean defaultLabel;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Milestone {
        private Long id;
        private Integer number;
        private String state;
        private String title;
        private String description;
        private GitHubUser creator;
        @JsonProperty("open_issues")
        private Integer openIssues;
        @JsonProperty("closed_issues")
        private Integer closedIssues;
        @JsonProperty("created_at")
        private LocalDateTime createdAt;
        @JsonProperty("updated_at")
        private LocalDateTime updatedAt;
        @JsonProperty("closed_at")
        private LocalDateTime closedAt;
        @JsonProperty("due_on")
        private LocalDateTime dueOn;
    }
}

File: Gixor-Mobile/app/src/main/java/com/cpy3f2/gixor_mobile/model/entity/Issue.kt
package com.cpy3f2.gixor_mobile.model.entity

import com.google.gson.annotations.SerializedName


data class Issue(
    val id: Long? = null,
    val number: Int? = null,
    val state: String? = null,
    val title: String? = null,
    val body: String? = null,
    val user: GitHubUser? = null,
    val labels: List<Label>? = null,
    val assignee: GitHubUser? = null,
    val assignees: List<GitHubUser>? = null,
    val milestone: Milestone? = null,
    val locked: Boolean? = null,

    val activeLockReason: String? = null,
    val comments: Int? = null,

    @SerializedName("closed_at")
    val closedAt: String? = null,

    @SerializedName("created_at")
    val createdAt: String? = null,

    @SerializedName("updated_at")
    val updatedAt: String? = null,
    val repository: GitHubRepository? = null
) {

    data class Label(
        val id: Long? = null,
        val name: String? = null,
        val description: String? = null,
        val color: String? = null,
        val defaultLabel: Boolean? = null
    )


    data class Milestone(
        val id: Long? = null,
        val number: Int? = null,
        val state: String? = null,
        val title: String? = null,
        val description: String? = null,
        val creator: GitHubUser? = null,

        @SerializedName("open_issues")
        val openIssues: Int? = null,

        @SerializedName("closed_issues")
        val closedIssues: Int? = null,

        @SerializedName("created_at")
        val createdAt: String? = null,

        @SerializedName("updated_at")
        val updatedAt: String? = null,

        @SerializedName("closed_at")
        val closedAt: String? = null,

        @SerializedName("due_on")
        val dueOn: String? = null
    )
}